agent/session.go

```diff
@@ -31,26 +31,26 @@ type session struct {
 	conn *grpc.ClientConn
 	addr string
 
-	agent     *Agent
-	sessionID string
-	session   api.Dispatcher_SessionClient
-	errs      chan error
-	messages  chan *api.SessionMessage
-	tasks     chan *api.TasksMessage
+	agent       *Agent
+	sessionID   string
+	session     api.Dispatcher_SessionClient
+	errs        chan error
+	messages    chan *api.SessionMessage
+	assignments chan *api.AssignmentsMessage
```

Comment:
I don't quite see the connection between removing `Assign` and needing to inform `Update` whether it is dealing with complete or incremental data. The code in this PR already seems to use `Update` for complete sets:

```
+			case api.AssignmentsMessage_COMPLETE:
 +				if err := a.worker.Assign(ctx, msg.UpdateTasks); err != nil {
```